azure/defaults.go

```diff
@@ -331,7 +338,13 @@ func GetDefaultWindowsImage(k8sVersion, runtime string) (*infrav1.Image, error)
 		return nil, errors.New("containerd image only supported in 1.22+")
 	}
 
-	skuID, err := getDefaultImageSKUID(k8sVersion, "windows", "2019")
+	if osVersion == "" {
+		osVersion = DefaultWindowsServerVersion
+	} else {
+		osVersion = strings.ReplaceAll(strings.ToLower(osVersion), "windows-", "")
```

Comment:
One thing I noticed: we're trimming the "windows" prefix here but then we end up using it as "os-version" anyways which gets converted back to `windows-osVersion` anyways: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2194/files#diff-b29d010635a6d1d4de222d1f49dfb5312909f7d391d5ada7ac131a84f6848b43L285

This would require just a bit of shuffling so feel free to not do this, but, if you wanted to make the code slightly cleaner we could just change `getDefaultImageSKUID(k8sVersion, os, osVersion string)` to ` getDefaultImageSKUID(k8sVersion, osVersion string)` and just pass in `osVersion` directly if it's set, else default `osVersion` to `"windows-" + DefaultWindowsServerVersion`. It would work for Ubuntu too, we just need to pass in `"ubuntu-" + osVersion` instead of two separate variables.

Reply:
I can try doing these changes.

Out of curiosity, would similar functionality be useful for ubuntu too (possibility for testing 22.04 if/when it arrives)?

Right now it looks like the ubuntu version selected is only based on the K8s version (unless explicitly set in the marketplace image reference).

Reply:
it could be, are you wondering if you should make the annotation generic? unless it's really easy to do both, I would keep this PR scoped to windows and we can add something similar for ubuntu later if the use case arises. As much as possible we don't want to encourage folks to use annotations as UX for specifying an OS version so let's keep it limited to where it's actually needed.

Reply:
The latest commit incorporates these suggestions. PTAL